app/src/main/java/com/ciandt/dragonfly/example/features/projectselection/ProjectSelectionActivity.kt

```diff
@@ -102,30 +111,30 @@ class ModelSelectionActivity : BaseActivity(), ModelSelectionContract.View {
         recyclerView.visibility = INVISIBLE
 
         messageIcon.setImageDrawable(ContextCompat.getDrawable(this, R.drawable.ic_error))
-        messageText.text = getString(R.string.model_selection_error_message)
+        messageText.text = getString(R.string.project_selection_error_message)
         messageContainer.visibility = VISIBLE
     }
 
-    override fun update(models: List<Model>) {
+    override fun update(projects: List<Project>) {
         loading.visibility = INVISIBLE
         messageContainer.visibility = INVISIBLE
 
-        this.models.clear()
-        this.models.addAll(models)
+        this.projects.clear()
+        this.projects.addAll(projects)
 
         recyclerView.adapter.notifyDataSetChanged()
 
         recyclerView.visibility = VISIBLE
     }
 
-    override fun update(model: Model) {
-        if (models.contains(model)) {
-            val index = models.indexOf(model)
-            models[index] = model
+    override fun update(project: Project) {
+        if (projects.contains(project)) {
+            val index = projects.indexOf(project)
```

Comment:
I need this index to notifyItemChanged on adapter.
We could have an method replace() that returns the index if it exists and -1 if not, but I would still have an if (index != -1) { //updated }  else { // insert }